app/src/main/java/com/example/news_app/fragment/SearchFragment.java

```diff
@@ -124,11 +138,12 @@ public void onSuccess(int statusCode, Headers headers, JSON json) {
                         Log.d(TAG, "onSuccess");
                         JSONObject jsonObject = json.jsonObject;
                         try {
-                            JSONArray items = jsonObject.getJSONArray("items");
+                            Type searchListType = new TypeToken<List<Search>>(){}.getType();
+                            List<Search> items = new Gson().fromJson(jsonObject.getJSONArray("items").toString(),searchListType);
```

Comment:
Try to do one function -> one purpose. Here, this logic can be split up into two smaller functions, parseSearchResponse() and updateResponseItems()